core/src/main/java/org/apache/brooklyn/core/location/internal/LocationConfigMap.java

```diff
@@ -53,10 +58,18 @@ public LocationConfigMap(AbstractLocation loc, Map<ConfigKey<?>, Object> storage
 
     @Override
     protected ExecutionContext getExecutionContext(BrooklynObject bo) {
-        if (bo==null) return null;
-        ManagementContext mgmt = ((AbstractLocation)bo).getManagementContext();
-        if (mgmt==null) return null;
-        return mgmt.getServerExecutionContext();
+        // Support DSL in location config. DSL expects to be resolved in the execution context of an entity.
+        // Since there's no location-entity relation try to infer it from the context of the caller.
+        Entity contextEntity = BrooklynTaskTags.getTargetOrContextEntity(Tasks.current());
+        if (contextEntity != null) {
+            return ((EntityInternal)contextEntity).getExecutionContext();
```

Comment:
@ahgittin Any thoughts on this change? It's retrieving the context entity's execution context if there's one, to be used as the location's execution context.
It's actually the same as what we currently do, just cleaning it up a bit.